Code/Framework/AzFramework/AzFramework/Asset/AssetCatalog.cpp

```diff
@@ -808,96 +808,118 @@ namespace AzFramework
-                }
-    #endif
+                    if (!isNewAsset && isCatalogInitialize)
+                    {
+                        // Nothing to do here - catalog initialize messages are intended to sync with the AP catalog.
+                        // Since this asset is already known, just stop processing to avoid triggering reloads.
+                        return;
+                    }
 
-                const AZ::Data::AssetType& assetType = isNewAsset ? message.m_assetType : assetInfoPair->second.m_assetType;
+#if defined(AZ_ENABLE_TRACING)
+                    if (message.m_assetType == AZ::Data::s_invalidAssetType)
+                    {
+                        AZ_TracePrintf(
+                            "AssetCatalog",
+                            "Registering asset \"%s\" via AssetSystem message, but type is not set.\n",
+                            relativePath.c_str());
+                    }
+#endif
 
-                AZ::Data::AssetInfo newData;
-                newData.m_assetId = assetId;
-                newData.m_assetType = assetType;
-                newData.m_relativePath = message.m_data;
-                newData.m_sizeBytes = message.m_sizeBytes;
+                    const AZ::Data::AssetType& assetType = isNewAsset ? message.m_assetType : assetInfoPair->second.m_assetType;
 
-                m_registry->RegisterAsset(assetId, newData);
-                m_registry->SetAssetDependencies(assetId, message.m_dependencies);
+                    AZ::Data::AssetInfo newData;
+                    newData.m_assetId = assetId;
+                    newData.m_assetType = assetType;
+                    newData.m_relativePath = message.m_data;
+                    newData.m_sizeBytes = message.m_sizeBytes;
 
-                for (const auto& mapping : message.m_legacyAssetIds)
-                {
-                    m_registry->RegisterLegacyAssetMapping(mapping, assetId);
+                    m_registry->RegisterAsset(assetId, newData);
+                    m_registry->SetAssetDependencies(assetId, message.m_dependencies);
+
+                    for (const auto& mapping : message.m_legacyAssetIds)
+                    {
+                        m_registry->RegisterLegacyAssetMapping(mapping, assetId);
+                    }
                 }
-            }
-            if (!isNewAsset)
-            {
-                // the following deliveries must happen on the main thread of the application:
-                AZ::SystemTickBus::QueueFunction([assetId]() 
+                if (!isNewAsset)
                 {
-                    AzFramework::AssetCatalogEventBus::Broadcast(&AzFramework::AssetCatalogEventBus::Events::OnCatalogAssetChanged, assetId);
-                });
+                    // the following deliveries must happen on the main thread of the application:
+                    AZ::SystemTickBus::QueueFunction(
```

Comment:
Just curious: Does it mean that all these queued notifications will be handled on the main thread (similar to TickEvents at https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/TickBus.h#L63) and is that expected?

Reply:
Yes, that is intended for this message, this isn't new code though and this branch won't run for my changes because of the check on line 834, this message isn't intended to trigger OnAssetReloaded.